uaclient/tests/test_data_types.py

```diff
@@ -170,6 +249,8 @@ class ExampleDataObject(DataObject):
         Field("integerlist_opt", data_list(IntDataValue), required=False),
         Field("objlist", data_list(ExampleNestedObject)),
         Field("objlist_opt", data_list(ExampleNestedObject), required=False),
+        Field("enum_opt_list", data_list(ExampleEnum), required=False),
+        Field("enum_opt", ExampleEnum, required=False),
```

Comment:
The intent of this example data object is to exercise every possible type. So we need some non optional enums and lists as well. Let's also continue the existing naming patterns for consistency.
```suggestion
        Field("enum", ExampleEnum),
        Field("enum_opt", ExampleEnum, required=False),
        Field("enumlist", data_list(ExampleEnum)),
        Field("enumlist_opt", data_list(ExampleEnum), required=False),

```